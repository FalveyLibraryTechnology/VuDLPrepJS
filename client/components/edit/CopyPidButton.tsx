import React from "react";
import ContentCopy from "@mui/icons-material/ContentCopy";

interface CopyPidButtonProps {
    pid: string;
}

const CopyPidButton = ({ pid }: CopyPidButtonProps): React.ReactElement => {
    return (
        <button onClick={() => navigator.clipboard.writeText(pid)}>
            <ContentCopy titleAccess={`Copy PID (${pid}) to clipboard`} />
        </button>
    );
};

export default CopyPidButton;
