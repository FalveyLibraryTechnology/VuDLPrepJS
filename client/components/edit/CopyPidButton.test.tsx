import React from "react";
import renderer from "react-test-renderer";
import CopyPidButton from "./CopyPidButton";

describe("CopyPidButton", () => {
    it("renders", async () => {
        const tree = renderer.create(<CopyPidButton pid="foo" />);
        expect(tree.toJSON()).toMatchSnapshot();
    });
});
