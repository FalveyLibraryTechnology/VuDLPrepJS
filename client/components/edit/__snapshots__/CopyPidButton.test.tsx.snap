// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyPidButton renders 1`] = `
<button
  onClick={[Function]}
>
  <svg
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="ContentCopyIcon"
    focusable="false"
    role="img"
    style={
      {
        "height": "14px",
      }
    }
    viewBox="0 0 24 24"
  >
    <path
      d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"
    />
    <title>
      Copy PID (foo) to clipboard
    </title>
  </svg>
</button>
`;
